pkg/cloudprovider/aws/amifamily/ami.go

```diff
@@ -109,16 +111,24 @@ func (p *AMIProvider) getAMIRequirements(ctx context.Context, providerRef *v1alp
 }
 
 func (p *AMIProvider) selectAMIs(ctx context.Context, amiSelector map[string]string) (map[string]scheduling.Requirements, error) {
+	var amis []*ec2.Image
 	ec2AMIs, err := p.fetchAMIsFromEC2(ctx, amiSelector)
 	if err != nil {
 		return nil, err
 	}
 	if len(ec2AMIs) == 0 {
 		return nil, fmt.Errorf("no amis exist given constraints")
 	}
+	if _, exists := amiSelector["aws-ids"]; exists {
+		amis = ec2AMIs
+	} else {
+		mostRecentAMI := getMostRecentAMI(ec2AMIs)
```

Comment:
This would limit the AMIs returned even if they have separate requirements. The most prevalent would be if the tag selector return both arm64 and x86 AMIs. Both would need to be returned from this func.  I think it makes sense to move this prioritization into the `Get` func above when the amiID is being added to the mapping of AMI ID -> InstanceTypes:

```
		for _, instanceType := range nodeRequest.InstanceTypeOptions {
			for amiID, requirements := range amiRequirements {
				if err := instanceType.Requirements().Compatible(requirements); err == nil {
					amiIDs[amiID] = append(amiIDs[amiID], instanceType)
				}
			}
		}
```

You'd need to modify the selectAMIs func to return more info than just the IDs to be able to do a creation timestamp sort within `Get`

Reply:
I have made some changes so that the most recent ami is chosen after it is found to be compatible